tests/framework/matcher/conditions.go

```diff
@@ -166,6 +194,31 @@ func (c conditionMatcher) FailureMessage(actual interface{}) string {
 		return format.Message(conditions, fmt.Sprintf("to find condition of type '%v' and status '%s' but got '%s'", c.expectedType, c.expectedStatus, conditionStatus))
 	}
 
+	if c.expectedMessage != "" {
+		return c.messageFailure(conditions)
+	}
+
+	if c.expectedReason != "" {
+		return c.reasonFailure(conditions)
+	}
+	return ""
+}
+
+func (c conditionMatcher) reasonFailure(conditions []interface{}) string {
+	conditionReason, err := c.getReasonForExpectedConditionType(conditions)
+	if err != nil {
+		return err.Error()
+	}
+	if conditionReason == "" {
+		if c.conditionCanBeMissing {
```

Comment:
Should be a separate `c.reasionCanBeMissing`?

Reply:
I don't think that's required, we don't have the same for messages can just use negative assertions like `ToNot` to assert the same IMHO.